tests/BaGetter.Core.Tests/Upstream/UpstreamAuthenticationTests.cs

```diff
@@ -0,0 +1,153 @@
+            {
+                Assert.Equal("Basic", r.Headers.Authorization.Scheme);
+                Assert.Equal("dXNlcjpwYXNzd29yZA==", r.Headers.Authorization.Parameter);
+            })
+            .Verifiable(Times.Once());
+
+        var client = services.GetRequiredService<HttpClient>();
+        var res = await client.GetFromJsonAsync<ServiceIndexResponse>("http://localhost/v3/index.json");
+    }
+
+    [Fact]
+    public async Task TestMirrorBearerAuthConfiguration()
+    {
+        var (services, mock) = SetupApp(opt =>
+        {
+            opt.Authentication = new MirrorAuthenticationOptions
+            {
+                Type = MirrorAuthenticationType.Bearer,
+                Token = "token"
+            };
+        });
+
+        mock.Protected()
+            .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
+            .ReturnsAsync(new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.OK,
+                Content = JsonContent.Create(TestServices)
+            })
+            .Callback<HttpRequestMessage, CancellationToken>((r, c) =>
+            {
+                Assert.Equal("Bearer", r.Headers.Authorization.Scheme);
+                Assert.Equal("token", r.Headers.Authorization.Parameter);
+            })
+            .Verifiable(Times.Once());
+
+        var client = services.GetRequiredService<HttpClient>();
+        var res = await client.GetFromJsonAsync<ServiceIndexResponse>("http://localhost/v3/index.json");
+    }
+
+    [Fact]
+    public async Task TestMirrorCustomAuthConfiguration()
+    {
+        var (services, mock) = SetupApp(opt =>
+        {
+            opt.Authentication = new MirrorAuthenticationOptions
+            {
+                Type = MirrorAuthenticationType.Custom,
+                CustomHeaders = new Dictionary<string, string>
+                {
+                    { "X-Auth", "value1" },
+                }
+            };
+        });
+
+        mock.Protected()
+            .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
+            .ReturnsAsync(new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.OK,
+                Content = JsonContent.Create(TestServices)
+            })
+            .Callback<HttpRequestMessage, CancellationToken>((r, c) =>
+            {
+                Assert.Equal("value1", r.Headers.GetValues("X-Auth").First());
+            })
+            .Verifiable(Times.Once());
+
+        var client = services.GetRequiredService<HttpClient>();
+        var res = await client.GetFromJsonAsync<ServiceIndexResponse>("http://localhost/v3/index.json");
+    }
+
+    private static (IServiceProvider serivces, Mock<HttpMessageHandler> mockHandler) SetupApp(Action<MirrorOptions> setupOptions)
+    {
+        Mock<HttpMessageHandler> mockHandler = new();
```

Comment:
Add `MockBehaviour.Strict` to make sure only what you mocked can be called. Also improves failed test messages. Unless of course there's a specific reason not to that I'm not seeing.